### ISSUE TITLE
Implement  equivalent to ZSH's zle accept-line for better integration with atuin

### ISSUE THREAD

**Issue by gergelyk** [nushell/nushell#11065]
### Related problem

https://github.com/atuinsh/atuin/issues/1392#issuecomment-1812748133

### Describe the solution you'd like

Allow for sending a command to the commandline and executing it immediately.

**Comment by fdncred** [comment]
Seems like someone could add `commandline --accept` or some such thing. There's also options to look at reedline's `send` events to see if you can send an enter after a commandline update.

**Comment by arcuru** [comment]
As one additional detail, for atuin it would need to be usable from within a keybinding. I know that sometimes keybindings have their own quirks so I'm just calling that out in case it applies here in Nushell.

**Comment by sholderbach** [comment]
Does the [`event: executehostcommand`](https://docs.rs/reedline/latest/reedline/enum.ReedlineEvent.html#variant.ExecuteHostCommand) fit the bill for that? This would not add to the internal history but allow execution based on a keybinding?

**Comment by fdncred** [comment]
@sholderbach from what i've looked at, executehostcommand will run things like fzf and put the output on the command line but not execute the returned results. Maybe I'm mistaken, but I thought the OP was looking for a way to execute the item that gets put on the command line buffer.

**Comment by ellie** [comment]
> I thought the OP was looking for a way to execute the item that gets put on the command line buffer.

That's correct - we'd need a way for the keybinding to immediately execute the result in the buffer

This would be the same (functionally) as `zle accept-line` for zsh

**Comment by fdncred** [comment]
I think this is going to have to be a change in reedline. This is what I've tried without success.

Creating keybindings with these events after commandline has run.
1. submit event
2. submitornewline event
3. enter event
4. pipeline with (char -i 13) in it

There's something that's preventing anything from running after commandline. I'd also add that I don't really like how commandline runs as it creates a new command line and doesn't update the existing one. That should be fixed too.

Ugh, I also found another bug in commandline where it only correctly uses --cursor-end the first time.

**Comment by mrdgo** [comment]
> I think this is going to have to be a change in reedline. This is what I've tried without success.
> 
> Creating keybindings with these events after commandline has run.
> 
>     1. submit event
> 
>     2. submitornewline event
> 
>     3. enter event
> 
>     4. pipeline with (char -i 13) in it
> 
> 
> There's something that's preventing anything from running after commandline. I'd also add that I don't really like how commandline runs as it creates a new command line and doesn't update the existing one. That should be fixed too.
> 
> Ugh, I also found another bug in commandline where it only correctly uses --cursor-end the first time.

that certain something was that `event::read()?` that blocks without checking the event queue. In my PR I added a check, if the event queue is not empty before blocking.

**Comment by sholderbach** [comment]
Closed by #16193
